app/src/main/java/com/practicum/playlistmaker/media/ui/MediaActivity.kt

```diff
@@ -1,21 +1,70 @@
 package com.practicum.playlistmaker.media.ui
 
 import android.os.Bundle
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
 import androidx.activity.enableEdgeToEdge
+import androidx.activity.OnBackPressedCallback
 import androidx.appcompat.app.AppCompatActivity
 import androidx.core.view.ViewCompat
 import androidx.core.view.WindowInsetsCompat
+import com.google.android.material.tabs.TabLayoutMediator
 import com.practicum.playlistmaker.R
+import com.practicum.playlistmaker.databinding.ActivityMediaBinding
 
 class MediaActivity : AppCompatActivity() {
+
+    private lateinit var binding: ActivityMediaBinding
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enableEdgeToEdge()
-        setContentView(R.layout.activity_media)
-        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
+
+        binding = ActivityMediaBinding.inflate(layoutInflater)
+        setContentView(binding.root)
+
+        setupWindowInsets()
+        setupViewPager()
+        setupBackButton()
+        setupBackPressHandler()
+    }
+
+    private fun setupWindowInsets() {
+        ViewCompat.setOnApplyWindowInsetsListener(binding.root) { v, insets ->
             val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
             v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
             insets
         }
     }
+
+    private fun setupViewPager() {
+        val adapter = MediaPagerAdapter(this)
+        binding.viewPager.adapter = adapter
+
+        TabLayoutMediator(binding.tabLayout, binding.viewPager) { tab, position ->
+            tab.text = when (position) {
+                0 -> getString(R.string.favorite_tracks)
+                1 -> getString(R.string.playlists)
+
+                else -> ""
```

Comment:
Здесь можно чуть упростить:
```suggestion
                else -> getString(R.string.playlists)
```